olp-cpp-sdk-core/src/network2/NetworkResponse.cpp

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright (C) 2019 HERE Europe B.V.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * License-Filename: LICENSE
+ */
+
+#include <olp/core/network2/NetworkResponse.h>
+
+using namespace olp::network2;
+
+bool NetworkResponse::GetIsCancelled() const { return is_cancelled_; }
+
+int NetworkResponse::GetStatus() const { return status_; }
+
+const std::string& NetworkResponse::GetError() const { return error_; }
+
+NetworkResponse& NetworkResponse::WithIsCancelled(bool is_cancelled) {
```

Comment:
Same here
```suggestion
NetworkResponse& NetworkResponse::Cancel(bool cancel) {
```

Reply:
I disagree about your suggestion. For me this name a bit confusing and implies that we cancel Response and leaving the object in some invalid state.

I would suggest WithCancelled() as it is more reflects what this method actually doing.
```suggestion
NetworkResponse& NetworkResponse::WithIsCancelled(bool is_cancelled) {
```

Reply:
WithCancelled sounds ok to me.